tutorials/running-a-mpi-cluster-within-lan/index.md

```diff
@@ -0,0 +1,196 @@
+
+```bash
+$ mkdir cloud
+```
+
+To export the ```cloud``` directory, you create an entry in ```/etc/exports```
+
+```bash
+$  cat /etc/exports
+/home/mpiuser/cloud *(rw,sync,no_root_squash,no_subtree_check)
+```
+Here, instead of ```*``` you can specifically give out the IP address to which you want to share this folder to. But, this will just make our job easier.
+
+* **rw**: This is to enable both read and write option. **ro** is for read-only.
+* **sync**: This applies changes to the shared directory only after changes are committed.
+* **no_subtree_check**: This option prevents the subtree checking. When a shared directory is the subdirectory of a larger filesystem, nfs performs scans of every directory above it, in order to verify its permissions and details. Disabling the subtree check may increase the reliability of NFS, but reduce security.
+* **no_root_squash**: This allows root account to connect to the folder.
+
+> Thanks to [Digital Ocean](https://www.digitalocean.com/community/tutorials/how-to-set-up-an-nfs-mount-on-ubuntu-12-04) for help with tutorial and explanations. Content re-used on account of Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License. For information, read [here](https://creativecommons.org/licenses/by-nc-sa/4.0/).
+
+After you have made the entry, run the following.
+
+```bash
+$ exportfs -a
+```
+
+Run the above command, every time you make a change to ```/etc/exports```.
+
+If required, restart the ```nfs``` server
+
+```bash
+$ sudo service nfs-kernel-server restart
+```
+
+### NFS-Client
+
+Install the required packages
+
+```bash
+$ sudo apt-get install nfs-common
+```
+
+Create a directory in the client's machine with the samename ```cloud```
+
+```bash
+$ mkdir cloud
+```
+
+And now, mount the shared directory like
+
+```bash
+$ sudo mount -t nfs master:/home/mpiuser/cloud ~/cloud
+```
+
+To check the mounted directories,
+
+```bash
+$ df -h
+Filesystem      		    Size  Used Avail Use% Mounted on
+master:/home/mpiuser/cloud  49G   15G   32G  32% /home/mpiuser/cloud
+```
+
+## Step 5: Running MPI Programs
+
+For consideration sake, let's just take a sample program, that comes along with MPICH2 installation package ```mpich2/examples/cpi```.
+
+If you want to compile your own code,
+
+```bash
+$ mpicc -o mpi_sample mpi_sample.c
+```
+
+First copy your executable into the shared directory ```cloud``` or better yet, compile your code within the NFS shared directory.
+
+```bash
+$ cd cloud/
+$ pwd
+/home/mpiuser/cloud
+```
+
+To run it only in your machine, you do
+
+```bash
+$ mpirun -np 2 ./cpi # No. of processes = 2
+```
+
+Now, to run it within a cluster, 
+
+```bash
+$ mpirun -np 5 -hosts client,localhost ./cpi 
+#hostnames can also be substituted with ip addresses.
+```
+
+Or specify the same in a hostfile and 
+
+```bash
+$ mpirun -np 5 --hostfile mpi_file ./cpi
+```
+
+This should spin up your program in all of the machines that your **master** is connected to. 
+
```

Comment:
Can you provide a section here to guide people to another tutorial? I generally try to do an "Up next" or something like that and then point the user to another tutorial (or give them a link back to the beginner tutorial page)